solr/solrj/src/java/org/apache/solr/common/util/JsonRecordReader.java

```diff
@@ -438,18 +442,19 @@ void walkObject() throws IOException {
       }
     }
 
-    private void addChildDoc2ParentDoc(Map<String, Object> record, Map<String, Object> values) {
+    private void addChildDoc2ParentDoc(Map<String, Object> record, Map<String, Object> values, String path) {
+      String trimmedPath = trimPath(path);
```

Comment:
I think "path" should be renamed "key", and the trimPath() should be done by the caller, and not be called trimPath either for that matter -- it takes the "leaf" or "suffix" part of the path, or perhaps you have an idea of a better name.